### DISCUSSION TITLE
Long-running service calls

### DISCUSSION THREAD

**Discussion by natiox** [Blazored/Typeahead#204] [Q&A]
We use Blazored Typeahead with a long-running service call.  When users type their first search query (John) and later change it (to Jo), the first result comes back and confuses them because it doesn't match their later change.  We implemented a sort of cancellation token on our wrapper component.  Is there possibly a better way for us to handle this?

Thanks for the awesome tool,
Nathan

**Comment by chrissainty** [comment]
This might be something we can improve in the component. Perhaps if we pass a cancellation token when we invoke the search method this can be used to cancel an existing call and stop this behaviour. 

Could you raise this as an issue so we can track it?
